### DISCUSSION TITLE
Git status notifications when using S3 as source

### DISCUSSION THREAD

**Discussion by stebje** [fluxcd/flux2#2705] [Q&A] [answered]
👋  Hi, cross-posting a question previously asked in the [#flux](https://cloud-native.slack.com/archives/CLAJ40HV3/p1651683749321249) Slack channel:

I’m setting up Flux with **GitHub / AWS EKS / S3**, and after some testing I have a question about [Git commit status notifications](https://fluxcd.io/docs/guides/notifications/#git-commit-status):

❓ **Does anyone know a solution (or workaround) for having a S3 Bucket as Source, while still sending a Git commit status notification back to GitHub? The use case is:**
1. User pushes commit(s) to the GitHub repo where `.yml` manifests are located
2. GitHub syncs the updated `.yml` files to a S3 bucket (e.g. via GitHub Actions)
3. Flux reconciles the changes pushed to the S3 bucket source above
4. Flux triggers notification back to GitHub, and the status is updated for the commit that was synced to S3

I’ve been testing this but I don’t see a way that this could be achieved without some custom solution, considering the git status notification relies on the commit SHA and this metadata is lost when syncing the manifest files from GitHub to S3.

Thanks in advance for any input :bow:

**Comment by stefanprodan** [comment, marked as answer]
There is no way to do this with Flux. Instead of commit status you could use an alert and send events to your own webhook and from there write your own logic.

**Reply by stebje** [reply]
Thanks @stefanprodan for your feedback. Indeed, a custom solution using alerts/notifications is an option. Some component keeps track of the linkage between `git` commit SHA <--> S3 bucket revision ID, and during the custom webhook receiver logic there's an API call back to github to update the commit status.

Thanks again 👍
